From b4154dc70a2fbab69c80c2809497c68434a0a2cf Mon Sep 17 00:00:00 2001
From: Kamil Debski <k.debski@samsung.com>
Date: Fri, 25 Jan 2013 06:29:57 -0300
Subject: [PATCH 0889/1083] [media] vb2: Add support for non monotonic
 timestamps

Not all drivers use monotonic timestamps. This patch adds a way to set the
timestamp type per every queue.
In addition, set proper timestamp type in drivers that I am sure that use
either MONOTONIC or COPY timestamps. Other drivers will correctly report
UNKNOWN timestamp type instead of assuming that all drivers use monotonic
timestamps.

Signed-off-by: Kamil Debski <k.debski@samsung.com>
Signed-off-by: Kyungmin Park <kyungmin.park@samsung.com>
Reviewed-by: Sylwester Nawrocki <s.nawrocki@samsung.com>
Acked-by: Hans Verkuil <hans.verkuil@cisco.com>
Signed-off-by: Sylwester Nawrocki <s.nawrocki@samsung.com>
Signed-off-by: Mauro Carvalho Chehab <mchehab@redhat.com>
(cherry picked from commit 6aa69f99b2ecc7f9b387fcf22d30e6601b58819f)

Signed-off-by: Simon Horman <horms+renesas@verge.net.au>

Conflicts:
	drivers/media/platform/blackfin/bfin_capture.c
	drivers/media/platform/davinci/vpbe_display.c
	drivers/media/platform/davinci/vpif_capture.c
	drivers/media/platform/davinci/vpif_display.c
	drivers/media/platform/s3c-camif/camif-capture.c
	drivers/media/platform/s5p-fimc/fimc-capture.c
	drivers/media/platform/s5p-fimc/fimc-lite.c
	drivers/media/platform/s5p-mfc/s5p_mfc.c
	drivers/media/platform/soc_camera/atmel-isi.c
	drivers/media/platform/soc_camera/mx2_camera.c
	drivers/media/platform/soc_camera/mx3_camera.c
	drivers/media/platform/soc_camera/sh_mobile_ceu_camera.c
	drivers/media/platform/vivi.c
	drivers/media/usb/pwc/pwc-if.c
	drivers/media/usb/stk1160/stk1160-v4l.c
	drivers/media/usb/uvc/uvc_queue.c
	drivers/media/v4l2-core/videobuf2-core.c
(cherry picked from commit da0d56abbcb51c66fb1e17ec1fb463498c321d54)

Signed-off-by: Ryo Kataoka <ryo.kataoka.wt@renesas.com>
---
 include/media/videobuf2-core.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/media/videobuf2-core.h b/include/media/videobuf2-core.h
index 7f4007ee..58e8d3bc 100644
--- a/include/media/videobuf2-core.h
+++ b/include/media/videobuf2-core.h
@@ -318,6 +318,7 @@ struct vb2_queue {
 	const struct vb2_mem_ops	*mem_ops;
 	void				*drv_priv;
 	unsigned int			buf_struct_size;
+	u32				timestamp_type;
 
 /* private: internal use only */
 	enum v4l2_memory		memory;
-- 
1.8.3.2

